aws/sdk/examples/dynamodb/src/bin/crud.rs

```diff
@@ -0,0 +1,457 @@
+        .map(|()| rng.sample(Alphanumeric))
+        .map(char::from)
+        .take(n)
+        .collect()
+}
+
+/// Create a new table. It's remotely possible the random table name exists.
+async fn create_table(client: &dynamodb::Client, table: &str, key: &str) {
+    let ad = AttributeDefinition::builder()
+        .attribute_name(key)
+        .attribute_type(ScalarAttributeType::S)
+        .build();
+
+    let ks = KeySchemaElement::builder()
+        .attribute_name(key)
+        .key_type(KeyType::Hash)
+        .build();
+
+    let pt = ProvisionedThroughput::builder()
+        .read_capacity_units(10)
+        .write_capacity_units(5)
+        .build();
+
+    match client
+        .create_table()
+        .table_name(table)
+        .key_schema(ks)
+        .attribute_definitions(ad)
+        .provisioned_throughput(pt)
+        .send()
+        .await
+    {
+        Ok(_) => println!(),
+        Err(e) => {
+            println!("Got an error creating the table:");
+            println!("{}", e);
+            process::exit(1);
+        }
+    }
+}
+
+/// For add_item and scan_item
+#[derive(Clone)]
+struct Item {
+    table: String,
+    key: String,
+    value: String,
+    first_name: String,
+    last_name: String,
+    age: String,
+    utype: String,
+}
+
+/// Add an item to the table.
+async fn add_item(client: &dynamodb::Client, item: Item) {
+    let user_av = AttributeValue::S(item.value);
+    let type_av = AttributeValue::S(item.utype);
+    let age_av = AttributeValue::S(item.age);
+    let first_av = AttributeValue::S(item.first_name);
+    let last_av = AttributeValue::S(item.last_name);
+
+    match client
+        .put_item()
+        .table_name(item.table)
+        .item(item.key, user_av)
+        .item("account_type", type_av)
+        .item("age", age_av)
+        .item("first_name", first_av)
+        .item("last_name", last_av)
+        .send()
+        .await
+    {
+        Ok(_) => println!(),
+        Err(e) => {
+            println!("Got an error adding item to table:");
+            println!("{}", e);
+            process::exit(1);
+        }
+    }
+}
+
+/// Query the table for an item matching the input values.
+async fn query(client: &dynamodb::Client, item: Item) {
+    let value = &item.value;
+    let key = &item.key;
+    let user_av = AttributeValue::S(value.to_string());
+    let userav = AttributeValue::S(value.to_string());
+    let type_av = AttributeValue::S(item.utype);
+    let age_av = AttributeValue::S(item.age);
+    let first_av = AttributeValue::S(item.first_name);
+    let last_av = AttributeValue::S(item.last_name);
+    let mut found_match = true;
+    let mut cond = HashMap::new();
+    cond.insert("#key".to_string(), key.to_string());
+    let mut expr = HashMap::new();
+    expr.insert(":value".to_string(), user_av);
+    match client
+        .query()
+        .table_name(item.table)
+        .set_key_condition_expression(Some("#key = :value".to_string()))
```

Comment:
is there a reason you used `set_` instead of just `key_condition_expression` et al? you wouldn't need the `Some` wrapping in that case

Reply:
I'll update the code.